docs/guides/managing_packages.md

```diff
@@ -0,0 +1,254 @@
+
+- Develop the needed packages independently from the agent, inside the local registry (packages folder) and later add them to an agent. This is the recommended method.
+
+## The agent structure
+
+The first thing a developer needs to do before they start writing code is setting up the project. Instead of starting from scratch, we recommend to use our developer template:
+
+```bash
+git clone git@github.com:valory-xyz/dev-template.git
+cd dev-template/
+make new_env
+pipenv shell
+```
+
+The previous commands will generate a virtual environment with {{open_autonomy}} installed, an empty local registry, some useful tools for checking packages and dummy tests. After this, in the same fashion git asks to initialize its configuration using `git config`, {{open_autonomy}} asks the user to set the user name, whether we are using a local or remote registry as well as the type of registry (IPFS by default). For example, initialize your registry running the following command:
+
+```bash
+autonomy init --reset --author john_doe --remote --ipfs --ipfs-node "/dns/registry.autonolas.tech/tcp/443/https"
+```
+
+Now we are in the position of creating our first agent:
+
+```bash
+aea create my_agent
+cd my_agent
+```
+
+If you explore the newly created `my_agent` directoy, you'll see it contains the following structure:
+
+```bash
+my_agent
+    ├── aea-config.yaml
+    ├── connections
+    ├── contracts
+    ├── protocols
+    ├── skills
+    └── vendor
+           ├── __init__.py__
+           └── open_aea
+                   └── protocols
+                         └── signing
+```
+
+So your new agent just contains the basic signing protocol from `open_aea` inside the `vendor` directory. Remember that you are not expected to modify these third party packages inside this directory. All your packages go inside `my_agent/connections`, `my_agent/contracts`, `my_agent/protocols` and `my_agent/skills`.
+
+
+## Adding a remote package to the agent
+
+Let's say we now want to add a new feature to this agent, for instance the ability to communicate using the Tendermint protocol. Since that package is already available on the Autonolas IPFS registry, we'll use the `add command` with the `--remote` flag:
+
+```bash
+autonomy add protocol valory/tendermint:0.1.0:bafybeihcnjhovvyyfbkuw5sjyfx2lfd4soeocfqzxz54g67333m6nk5gxq --remote
+```
+
+You can check your new protocol package at the following location:
+
+```bash
+my_agent
+    └── vendor
+           └── valory
+                   └── protocols
+                           └── tendermint
+```
+
+It is useful to add these third party packages to the `.gitignore` so your repository just tracks your own packages.
+
+After adding a vendor package, you can remove it from the agent by:
+```bash
+autonomy eject protocol valory/tendermint:0.1.0
```

Comment:
No, eject doesn't do that. Eject moves a package from vendor into non-vendor area of the agent.

Reply:
push / publish is for pushing / publishing to local registry